projects/ion/src/lib/picker/date-picker/date-picker-calendar/date-picker-calendar.component.ts

```diff
@@ -67,39 +60,91 @@ export class IonDatePickerCalendarComponent implements OnInit, DoCheck {
     nextMonth: (): void => this.nextMonth(),
     nextYear: (): void => this.nextYear(),
   };
+  public finalRange = true;
 
   constructor() {
     this.setLanguage();
   }
 
-  setLanguage(): void {
-    if (!this.lang) {
-      this.lang = window.navigator.language;
-    }
-  }
-
   ngOnInit(): void {
     this.setCalendarInitialState();
     this.tempRenderDays();
   }
 
-  setCalendarInitialState(): void {
-    this.selectedDay = new Day(this.getInitialDate(), this.lang);
-    this.calendar = this.getCalendarInstance();
+  ngDoCheck(): void {
+    if (this.calendarControlAction) {
+      this.calendarAction[this.calendarControlAction]();
+    }
+  }
+
+  handleClick(dayIndex: number): void {
+    if (this.rangePicker) {
+      if (!this.selectedDay.length || this.selectedDay[FINAL_RANGE]) {
+        this.selectedDay = [this.days[dayIndex]];
+        return;
+      }
+
+      this.selectedDay[FINAL_RANGE] = this.days[dayIndex];
+      this.arrangeDates();
+      this.emitEvent();
+      this.setDateInCalendar();
+      return;
+    }
+    this.selectedDay = [this.days[dayIndex]];
+    this.emitEvent();
+    this.setDateInCalendar();
   }
 
-  getInitialDate(): Date {
-    return this.currentDate
-      ? new Date(this.currentDate.replace('-', ','))
-      : new Date();
+  isToday(date: Day): boolean {
+    const TODAY = new Day(new Date(), this.lang);
+    return isSameDay(date, TODAY);
   }
 
-  getCalendarInstance = (): Calendar =>
-    new Calendar(
-      this.selectedDay.year,
-      this.selectedDay.monthNumber,
-      this.lang
+  isSelectedDate(date: Day, isFinalOfRange?: boolean): boolean {
+    return isSameDay(
+      date,
+      isFinalOfRange
+        ? this.selectedDay[FINAL_RANGE]
+        : this.selectedDay[INITIAL_RANGE]
     );
+  }
+
+  isBetweenRange(date: Day): boolean {
+    if (this.selectedDay[INITIAL_RANGE] && this.selectedDay[FINAL_RANGE]) {
+      const [INITIAL_DATE, FINAL_DATE, CURRENT_DATE] = [
+        new Date(this.selectedDay[INITIAL_RANGE].Date),
```

Comment:
Acredito que você pode usar diretamente o `this.selectedDay[INITIAL_RANGE].Date` pois, o valor dele gera o mesmo resultado do  `new Date(this.selectedDay[INITIAL_RANGE].Date)`. Assim, não precisa fazer o `new Date`.

Reply:
Outra questão, essa função é chamada muitas vezes.